discovery_engine/lib/src/api/events/engine_events.dart

```diff
@@ -245,6 +246,16 @@ class EngineEvent with _$EngineEvent {
     SearchFailureReason reason,
   ) = SearchTermRequestFailed;
 
+  @Implements<SearchEngineEvent>()
+  const factory EngineEvent.trendingTopicsRequestSucceeded(
+    List<TrendingTopic> topics,
+  ) = TrendingTopicsRequestSucceeded;
+
+  @Implements<SearchEngineEvent>()
+  const factory EngineEvent.trendingTopicsRequestFailed(
+    SearchFailureReason reason,
```

Comment:
there is a `SearchFailureReason.noResultsAvailable` variant which might be enough for this case, otherwise we can add additional, trending topics specific variant(s)